### PR TITLE
[WIP] Script and API link updates due to angular2 moving to `dart-lang/angular`

### LINKED ISSUE #226: [ng] eliminate build script dependency on repo alias angular-dart -> angular2
Some build scripts copied from ng.io assumed that the name of the Angular Dart repo was `angular-dart`. Currently, the script that fetches the `angular2` repo also creates the `angular-dart` alias. Eventually, we should just drop the alias from all scripts.

### PR DESCRIPTION
**Pull request by chalin**
Note that this PR **does not** rename the `angular2` package in the docs (since that is still the official package name under pub).

In this PR:
- Adjust build scripts so that they pickup the `dart-lang/angular` repo (since `angular2` is now just redirected to `angular`).
- The Angular API reference is now created from `dart-lang/angular`.
- Updated links to API entries to refer to `angular` instead of `angular2`.

Staged at https://angulardart-org-dev.firebaseapp.com

Contributes to #786
Fixes #226

### CODE REVIEW

File: gulp/dartdoc.js [hunk 1]
```diff
@@ -10,18 +10,17 @@ module.exports = function (gulp, plugins, config) {
...
+      angular.compiler
+      angular.core
+      angular.platform.browser
+      angular.platform.common
+      angular.platform.common_dom
+      angular.router
+      angular.security
+      angular.transform.deferred_rewriter.dart
+      angular.transform.reflection_remover.dart
+      angular.transformer_dart`.replace(/\s+/g, ','),
```
Comment (kwalrath) [inline gulp/dartdoc.js]: The removal of browser_static made me look at the packages listed on https://webdev.dartlang.org/angular/api that aren't listed here. They include angular.platform.common_dom & angular.transform*. Do we still need those?

Comment (chalin) [inline gulp/dartdoc.js]: Good question. I'm not sure. @matanlurey will certainly have something to say about that.

Comment (chalin) [inline gulp/dartdoc.js]: `angular.trasform*` libraries don't have much documentation, and essentially redirect the reader to the wiki Transformer page (which doesn't exist anymore). So maybe we _should_ just drop the `angular.transform*` libraries.

**Comment by chalin** [comment]
We'll accumulate 4.x PRs over the `4-dev` branch -- see #805.

**Pull request closed without merging**